src/lib/mockserver.js

```diff
@@ -36,13 +37,20 @@ async function waitUntilReady(container) {
     let count = 1;
     while (count <= 10) {
         await sleep(100 * count); // to account for cold starts
-        const logs = await exec(`docker logs ${container}`);
+        let logs;
+        try {
+            logs = await exec(`docker logs ${container}`, { maxBuffer: MAX_BUFFER_SIZE});
+        // eslint-disable-next-line no-unused-vars
+        } catch (e) {
+            throw new Error(`Log output of ${container} too large`);
+        }
         const portIsRunning = logs.stdout.includes('started on ports: [80, 443]');
```

Comment:
Hmm, for the purpose of finding out if the ports are ready, there should actually be a better way than looking at the log files. It seems pretty wasteful and possibly slow.

Do you remember if mock-server can handle request once ports are up? In which case `docker ports` + pinging the port would help? Or does it open the ports but still does internal initialization until that log message is there?

If the log approach is the only one: Also we can use `spawn` instead of `exec` and get stdout (logs) line by line and never have to buffer all of it. The retry logic would then change into a `data` listener [like here](https://github.com/adobe/aio-cli-plugin-asset-compute/blob/master/src/lib/actionrunner.js#L329-L336) which would wait until it sees "started on ports: [80, 443]" and if yes, resolve the promise (and kill the spawned process). For the timeout, there would be a `setTimeout()` that would reject after X min (and kill the spawned process).

Reply:
IIRC, I think it opens up the ports right away but still needs to set up the mocks internally, hence why I had to wait until the logs include a message.

I will check this.

btw, how do I run the unit tests locally? They keep failing even though they pass on travis ci

Reply:
(What is the unit test errors you see? Just that it failed or does it give more info?)

Reply:
So turns out the tests were failing because I had some old docker containers from manual testing left running. I removed them and got the tests to work finally on my own machine.

 I just pushed my changes using `spawn` instead of `exec`. Also, I could not figure out how to use `setTimeout()` instead of a while loop with a counter. I spent a long time going around in circles with unresolved promises.

Reply:
Did you need to use `resolve` and `reject` functions from `new Promise((resolve, reject) => ...` instead of `Promise.reject()`/`Promise.resolve()`? The latter only work in certain cases I think.

Reply:
I did the former. What happened was `proc.stdout.on` resolved multiple times but it still had to wait the full 10s to stop the function.

Btw, I also just tried the latter and it resulted in an unresolved promise